### DISCUSSION TITLE
VSCode extension context menu placement.

### DISCUSSION THREAD

**Discussion by rodelta** [pieces-app/support#339] [Q&A] [answered]
Just installed the VSCode extension, and pieces added options in the context menu at the top of the list. 
This list is reserved for navigation items, and traditionally extensions add their options further down the list. The current positioning gets in the way of creating files and folders, which are used very frequently.

In my opinion, by default, the options should be further down the list. 

Is there a way to disable or move these items in the context menu without having to set up an override for vscode? I searched docs and online, and couldn't find anything relevant.

![CleanShot 2024-08-01 at 13 56 24](https://github.com/user-attachments/assets/8ae8affa-074e-429a-a507-6995b07015d2)

**Comment by wilfriedago** [comment]
I'm thinking the same thing, I miss creating a new file many times, just because those actions are stuck to the top, and this kinda disturbs my workflow.

**Comment by wilfriedago** [comment, marked as answer]
Hey @rodelta, I found a workaround to kinda move those actions down, both in the explorer and the editor context menu.

If you're looking to change the position of context menu actions for an extension in VS Code, here’s a step-by-step guide on how to do it. This example specifically addresses the `pieces-vscode` extension, but the steps can be adapted for other extensions as well.

#### Steps:

1. **Locate the Extension Directory:**
   - Navigate to the VS Code extensions folder. On macOS, this is typically found at `~/.vscode/extensions`.
   - Locate the specific extension directory, which will be named something like `meshintelligenttechnologiesinc.pieces-vscode-${version}`. Note that the `${version}` part will vary based on the installed version of the extension.

2. **Edit the `package.json` File:**
   - Inside the extension directory, find and open the `package.json` file.

3. **Modify the Menu Grouping:**
   - Adjust the `group` attribute for the context menu actions you want to move. The `group` attribute determines the order in which the actions appear in the menu. Here’s an example of what the `package.json` modifications might look like:

   **Original:**
   ```json
   ...
   "editor/context": [
       {
           "submenu": "pieces.pieces",
           "group": "navigation"
       }
   ],
   "explorer/context": [
       {
           "command": "pieces-vscode.askAboutFile",
           "group": "navigation"
       },
       {
           "command": "pieces-vscode.addAnchorToContext",
           "group": "navigation"
       },
       {
           "command": "pieces-vscode.saveFileToPieces",
           "group": "navigation",
           "when": "resourceLangId && !explorerResourceIsFolder"
       }
   ],
   "terminal/context": [
       {
           "command": "pieces-vscode.askQGPTTerminal",
           "group": "navigation"
       }
   ],
   ...
   ```

   **Modified:**
   ```json
   ...
   "editor/context": [
       {
           "submenu": "pieces.pieces",
           "group": "1_modification"
       }
   ],
   "explorer/context": [
       {
           "command": "pieces-vscode.askAboutFile",
           "group": "2_workspace"
       },
       {
           "command": "pieces-vscode.addAnchorToContext",
           "group": "2_workspace"
       },
       {
           "command": "pieces-vscode.saveFileToPieces",
           "group": "2_workspace",
           "when": "resourceLangId && !explorerResourceIsFolder"
       }
   ],
   "terminal/context": [
       {
           "command": "pieces-vscode.askQGPTTerminal",
           "group": "3_edit"
       }
   ],
   ...
   ```

   In this example, we’ve moved the actions to different groups to control their order in the context menus.

4. **Save and Reload:**
   - Save your changes to the `package.json` file.
   - Reload VS Code or the specific extension to see the changes take effect.

#### Additional Information:
You can find more details about the available sorting groups in the [VS Code API documentation](https://code.visualstudio.com/api/references/contribution-points#Sorting-of-groups).

**Comment by caleb-at-pieces** [comment]
Hey guys, we updated the context menu to hopefully avoid this issue and have a better experience. Thanks for the discussion!
